pybop/models/base_model.py

```diff
@@ -803,3 +803,7 @@ def spatial_methods(self):
     @property
     def solver(self):
         return self._solver
+
+    @solver.setter
+    def solver(self, solver):
+        self._solver = solver.copy() if solver is not None else None
```

Comment:
Q: Is it safe to update this property both before and after a model is built?

Reply:
It should be, I haven't ran into any issues modifying it, as the solver is independent of the built model. This also follows PyBaMM's Simulation class implementation of the same property.